text/0043-disable-sbom.md

```diff
@@ -0,0 +1,50 @@
+# Provide Global Mechanism to Disable SBOM Generation
+
+## Summary
+
+Users should have a mechanism to disable the generation of SBOM documents
+during the build process.
+
+## Motivation
+
+The generation of the SBOM documents for an image can be very time-intensive in
+some cases. We have seen a doubling of build times for Node.js applications.
+Additionally, some users are not interested in taking advantage of the SBOM
+features of the buildpacks.
+
+## Detailed Explanation
+
+If a user wishes to produce an image that does not contain SBOM documents, they
+can set the `BP_GENERATE_SBOM` environment variable to `false`. When unset, the
+value of this environment variable is assumed to be `true`.
+
+When `BP_GENERATE_SBOM` is set to `false`, none of the buildpacks executing
+during the build phase should generate or attach an SBOM in their outputs. This
+would apply to both new (Syft, CycloneDX, and SPDX formats) and old (label) SBOM
+outputs.
+
+## Rationale and Alternatives
+
+Alternatively, we could not provide an mechanism to disable SBOM generation.
+This does not seem like a great option as the build-time penalty introduced by
+SBOM generation can be substantial.
+
+We could also not specify anything at the organizational level and let each
+buildpack subteam decide how they would like to handle this for their
+ecosystem. However, with regards to this particular feature, it doesn't seem
+like there would be a big need to have differing implementations of this flag
+for each ecosystem. It would mean in some polyglot build scenarios that users
+might need to specify that they'd like to disable SBOM generation in more than
+one way. We and users would likely benefit more from a standardized feature
+flag that worked uniformly across all buildpacks.
+
+## Implementation
+
+All buildpacks that generate an SBOM will be required to honor the
+`BP_GENERATE_SBOM` environment variable. When set to `false` these buildpacks
```

Comment:
```suggestion
A buildpack that generates an SBOM and wishes to allow users to skip SBOM generation, should honor the `BP_GENERATE_SBOM` environment variable. When set to `false` these buildpacks
```

Thoughts about making this optional? I can see where it's useful for some buildpacks. If SBOM generation adds a perceptible amount of delay, allowing users to skip would be helpful. 

On the Java side, the scan is very fast. I'm not sure we'd want to implement this unless something changes and the scanning starts to impact build times.

Making this option would ensure the UX is consistent across language families if a language family decides to permit this.

Reply:
Personally I think it's more confusing to have some buildpacks in the project (cross-language) that allow it to be disabled and other that do not. If the default is to generate the SBOM, I don't see much downside to allowing users to disable it.

Reply:
I see value in the SBOM being consistently present on all images we generate. It may not matter at dev time, but for production environments, I think adoption in larger companies could be slowed if we allow this to be disabled. Assuming we go ahead with this RFC, making the implementation of it optional would still allow us to still make that guarantee for some language families, where it's feasible.

Reply:
I've changed the language here to specify that this is opt-in behavior for a buildpack.